Networking/Networking/Mapper/ProductTagListMapper.swift

```diff
@@ -18,18 +18,38 @@ struct ProductTagListMapper: Mapper {
         decoder.userInfo = [
             .siteID: siteID
         ]
+        let hasDataEnvelope = response.hasDataEnvelope
 
         switch responseType {
-          case .load:
-            do {
+        case .load:
+            if hasDataEnvelope {
                 return try decoder.decode(ProductTagListEnvelope.self, from: response).tags
-            } catch {
+            } else {
                 return try decoder.decode([ProductTag].self, from: response)
             }
-          case .create:
-            return try decoder.decode(ProductTagListBatchCreateEnvelope.self, from: response).tags
-          case .delete:
-            return try decoder.decode(ProductTagListBatchDeleteEnvelope.self, from: response).tags
+        case .create:
+            let tags: [ProductTagFromBatchCreation] = {
+                if hasDataEnvelope {
+                    return (try? decoder.decode(ProductTagListBatchCreateEnvelope.self, from: response).data.tags) ?? []
+                } else {
+                    return (try? decoder.decode(ProductTagListBatchCreateContainer.self, from: response).tags) ?? []
```

Comment:
❓ If there's an error while decoding the response, it looks like it would silently fail here instead of throwing the error? maybe a do/catch is still necessary for the processing after decoding the tags. alternatively, separating this into a different function that throws might also work.

Reply:
I was following the [original implementation](https://github.com/woocommerce/woocommerce-ios/blob/32e91b3f4bae83a44cd1f14c34e53a21e526e9cb/Networking/Networking/Mapper/ProductTagListMapper.swift#L68-L75) - but it makes sense to make it fail so we can log errors. I see that we do catch errors to show on alerts [here](https://github.com/woocommerce/woocommerce-ios/blob/32e91b3f4bae83a44cd1f14c34e53a21e526e9cb/WooCommerce/Classes/ViewRelated/Products/Edit%20Product/Edit%20Tags/ProductTagsViewController.swift#L252-L254) - so I removed the silent failures in a66278f8d9cb907dd1b05fbfbba7c77996f1c5ff.